backend/src/board.rs

```diff
@@ -115,21 +106,25 @@ impl Board {
         let logical = layout.layout.values().next().unwrap();
```

Comment:
Ya you are right about unwrapping a mutex. In situations like this where the function return is already a `Result_, String>` returning a result is easy enough that I don't think we should have unwrap calls here,

Reply:
But on second review most situations like this are not a result of this PR.

Reply:
I will approve if you want to, but seems like a simple enough change

Reply:
I'd say panicking is the correct behavior if the `Layout` is invalid. `layout` here can only ever be one of the layouts compiled into the library. Libraries should not attempt to return errors for things that are internal bugs in the library itself.

Reply:
I disagree. 
If a function returns a result if the function is called incorrectly, then it should return an `Err`, not panic, that is the point of the `Result` enum.